### ISSUE TITLE
log4j2-jboss-logmanager LevelTranslator NullPointer

### ISSUE THREAD

**Issue by jamezp** [jboss-logging/log4j2-jboss-logmanager#3]
We found the following Nullpointer Exception
```
Caused by: java.lang.NullPointerException
	at org.jboss.logmanager.log4j.LevelTranslator.translateLevel(LevelTranslator.java:95)
	at org.jboss.logmanager.log4j.JBossLogger.getLevel(JBossLogger.java:156)
```
....
Reason seems that LevelTranslator is not defensive enough. Suggestion:
```
    java.util.logging.Level translateLevel(final Level level) {
        //level null is same as level not translated
    	if(level == null) {
        	return org.jboss.logmanager.Level.INFO;
        }
    	final java.util.logging.Level result = log4jToJul.get(level.intLevel());
        return result == null ? org.jboss.logmanager.Level.INFO : result;
    }

    Level translateLevel(final java.util.logging.Level level) {
     //level null is same as level not translated        
        if(level == null) {
        	return Level.INFO;
        }
    	final Level result = julToLog4j.get(level.intValue());
        return result == null ? Level.INFO : result;
    }
```

Copied from https://issues.redhat.com/browse/LOGMGR-266.

**Comment by boris-unckel** [comment]
There might be a second point:
org.jboss.logmanager.log4j.JBossLogger
` @Override
    public Level getLevel() {
    	final java.util.logging.Level julLevel = logger.getLevel();
    	// Fallback to EffectiveLevel if underlying java.util.logging.Logger's level is null 
    	int julLevelEffectiveInt;
    	if(julLevel == null) {
    		julLevelEffectiveInt = logger.getEffectiveLevel();
    	} else {
    		julLevelEffectiveInt = julLevel.intValue();
    	}
        return levelTranslator.translateLevel(julLevelEffectiveInt);
    }`
and the translation by int in org.jboss.logmanager.log4j.LevelTranslator
` Level translateLevel(final int julLevelInt) {
    	final Level result = julToLog4j.get(julLevelInt);
        return result == null ? Level.INFO : result;
    }`

**Comment by jamezp** [comment]
How does this look @boris-unckel? https://github.com/jboss-logging/log4j2-jboss-logmanager/pull/4

**Comment by boris-unckel** [comment]
Great, includes both points. Do you plan to put log4j2-jboss-logmanager into the next wildfly-core release?

**Comment by jamezp** [comment]
Well that's a good question :) What is currently here now would work, but in WildFly a user would not be able to use a log4j2 configuration file without excluding the dependency from their deployment.

I was thinking about adding the ability to include `org.apache.logging.log4j:log4j-core` in a deployment which would take over the log manager responsibilities from the `org.jboss.logmanager:jboss-logmanager`. This would allow log4j2 configurations to be used. However I'm not sure how well it will work. I've got it working somewhat locally, but haven't finished testing it yet.

Any insight on how you'd expect log4j2 to work in WildFly is welcome :) FWIW the WildFly issue is https://issues.redhat.com/browse/WFCORE-482.

**Comment by jamezp** [comment]
Resolved with #4
